### PR TITLE
Expand templates in checksum keys

### LINKED ISSUE #22090: Failure due to templating in checksums of extension
Test are failing in #22088 for `Perl-bundle-CPAN-5.36.1-GCCcore-12.3.0.eb` due to the usage of a `%(version)s` template in the checksum specification of an extension:

```
== 2024-12-21 00:31:11,868 easyblock.py:871 INFO Found file Sys-Info-Driver-Linux-0.7905.tar.gz at /project/def-maintainers/boegelbot/sources/p/Perl-bundle-CPAN/extensions/Sys-Info-Driver-Linux-0.7905.tar.gz
== 2024-12-21 00:31:11,868 filetools.py:1265 DEBUG Using blocksize 16777216 for calculating the checksum
== 2024-12-21 00:31:11,869 easyblock.py:692 INFO sha256 checksum for /project/def-maintainers/boegelbot/sources/p/Perl-bundle-CPAN/extensions/Sys-Info-Driver-Linux-0.7905.tar.gz: 899c329bd3508ec5849ad0e5dadfa7c3679bbacaea9dda12404a7893032e8b7b
== 2024-12-21 00:31:11,869 easyblock.py:695 DEBUG Verifying checksums for extension source...
== 2024-12-21 00:31:12,039 build_log.py:226 ERROR EasyBuild encountered an error (at easybuild/5.0.x/easybuild-framework/easybuild/base/exceptions.py:126 in __init__): Missing checksum for Sys-Info-Driver-Linux-0.7905.tar.gz in {'Sys-Info-Driver-Linux-%(version)s.tar.gz': '899c329bd3508ec5849ad0e5dadfa7c3679bbacaea9dda12404a7893032e8b7b'} (at easybuild/5.0.x/easybuild-framework/easybuild/tools/filetools.py:1307 in verify_checksum)
```

This is the easyconfig code related to this error:
https://github.com/easybuilders/easybuild-easyconfigs/blob/ef644ae1e1edeab766f0c9632bdd8a210186737e/easybuild/easyconfigs/p/Perl-bundle-CPAN/Perl-bundle-CPAN-5.36.1-GCCcore-12.3.0.eb#L2120-L2135

### PR DESCRIPTION
**Pull request by Micket**
There are more cases for things that do nested dicts with arch specific binary source tarballs. I started changing those as well, but there were a lot and it seems to work anyway, so i'm not sure if we should "fix" those as well.
They are all cases where a normal --inject-checksums wouldn't work (due to multiple source variants).

edit: fixes #22090

### CODE REVIEW

File: easybuild/easyconfigs/c/CellBender/CellBender-0.2.1-foss-2021a-CUDA-11.3.1.eb [hunk 1]
```diff
@@ -38,7 +38,7 @@ exts_list = [
         'sources': ['v%(version)s.tar.gz'],
         'patches': ['CellBender-0.2.1-sphinxremoval.patch'],
         'checksums': [
-            {'v%(version)s.tar.gz':
+            {'0.2.1.tar.gz':
```
Comment (lexming) [inline easybuild/easyconfigs/c/CellBender/CellBender-0.2.1-foss-2021a-CUDA-11.3.1.eb:41]:
```suggestion
            {'v0.2.1.tar.gz':
```

File: easybuild/easyconfigs/p/Perl-bundle-CPAN/Perl-bundle-CPAN-5.36.1-GCCcore-12.3.0.eb [hunk 2]
```diff
@@ -2056,22 +2056,6 @@ exts_list = [
         'source_urls': ['https://cpan.metacpan.org/authors/id/W/WS/WSNYDER'],
         'checksums': ['3973ebdc44682c9c15c776f66e8be242cb4ff1dd52caf43ff446b74d4dccca06'],
     }),
-    ('Sys::Info::Driver::Linux::Device::CPU', '0.7905', {
```
Comment (boegel) [inline easybuild/easyconfigs/p/Perl-bundle-CPAN/Perl-bundle-CPAN-5.36.1-GCCcore-12.3.0.eb:2059]: Hmm, we should probably retain the top entry here, not the bottom one...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 27 out of 27 (27 easyconfigs in total)
node3509.doduo.os - Linux RHEL 8.8, x86_64, AMD EPYC 7552 48-Core Processor, Python 3.6.8
See https://gist.github.com/boegel/bb13c02dcf4915cfce1955fd3a1d1aec for a full test report.

File: easybuild/easyconfigs/p/Perl-bundle-CPAN/Perl-bundle-CPAN-5.36.1-GCCcore-12.3.0.eb (hunk 2 above)
Comment (boegel) [reply easybuild/easyconfigs/p/Perl-bundle-CPAN/Perl-bundle-CPAN-5.36.1-GCCcore-12.3.0.eb:2059]: Nevermind, same order as in other `Perl-bundle-CPAN` easyconfigs, so should be fine

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node3509.doduo.os - Linux RHEL 8.8, x86_64, AMD EPYC 7552 48-Core Processor, Python 3.6.8
See https://gist.github.com/boegel/f49355993ea069c0af260d37c9f69ada for a full test report.

File: easybuild/easyconfigs/p/Perl-bundle-CPAN/Perl-bundle-CPAN-5.36.1-GCCcore-12.3.0.eb (hunk 2 above)
Comment (boegel) [reply easybuild/easyconfigs/p/Perl-bundle-CPAN/Perl-bundle-CPAN-5.36.1-GCCcore-12.3.0.eb:2059]: Ah, no, order really does matter, installation of `Sys-Info` is now failing with:
```
Can't locate Sys/Info/Driver/Linux/Device/CPU.pm in @INC
```

I'll fix that in this PR...

Comment (boegel) [reply easybuild/easyconfigs/p/Perl-bundle-CPAN/Perl-bundle-CPAN-5.36.1-GCCcore-12.3.0.eb:2059]: fixed in 9d8f5e25e1

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3509.doduo.os - Linux RHEL 8.8, x86_64, AMD EPYC 7552 48-Core Processor, Python 3.6.8
See https://gist.github.com/boegel/b40d719cde503292ba71b5184ff27fb5 for a full test report.

**Review by boegel — APPROVED** [review]

**Pull request merged**